### PR TITLE
fix: picker an arrow navigation

### LINKED ISSUE #3858: Emoji picker - Unable to navigate with arrow keys when there's text in search box
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Navigate to e.cash 
2. Navigate to a conversation
3. Click on the emoji icon
4. Type something in the search box
5. Press down arrow key to navigate through the options 

## Expected Result:
User should be able to navigate the emoji list with arrow keys after pressing the down arrow

## Actual Result:
Pressing the down arrow does not allow the user to navigate through the emoji list 

## Workaround:
User has to manually select the emoji with the mouse cursor

## Platform:
Where is this issue occurring?

 Web ✔️ 
 iOS
 Android
 Desktop App ✔️ 
 Mobile Web

**Version Number:** 1.0.74-0

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/44479856/124185045-b7f52280-daba-11eb-80f8-9826d6a80bc1.mp4

**Expensify/Expensify Issue URL:**

[View all open jobs on Upwork](https://www.upwork.com/ab/jobs/search/?q=Expensify%20React%20Native&sort=recency&user_location_match=2)

___ 
From @mallenexpensify https://expensify.slack.com/archives/C01GTK53T8Q/p1625155126323200?thread_ts=1625154887.322400&cid=C01GTK53T8Q

> With cursor in compose box, click tab > enter
> Start typing a search for an emoji. 
> Observe you can't click the down arrow to then navigate the emoji selector to choose which emoji you want.

### PR DESCRIPTION
**Pull request by parasharrajat**
### Details

1. https://github.com/Expensify/Expensify.cash/issues/3858#issuecomment-873079314
2. https://github.com/Expensify/Expensify.cash/issues/3858#issuecomment-873122883

### Fixed Issues

$ Fixes #3858

### Tests | QA Steps

1. Navigate to e.cash
2. Navigate to a conversation
3. Click on the emoji icon
4. Type something in the search box
5. Press the down arrow key to navigate through the options.
6. when at the top of the emoji list, press the Up arrow to focus the search Input.

### Tested On

- [x] Web (Affected Platform)
- [ ] Mobile Web
- [x] Desktop (Affected Platform)
- [ ] iOS
- [ ] Android

#### Web

https://user-images.githubusercontent.com/24370807/124370305-9a2ee580-dc93-11eb-90af-8c96ccdda73e.mp4

### CODE REVIEW

File: src/pages/home/report/EmojiPickerMenu/index.js [hunk 1]
```diff
@@ -110,7 +105,9 @@ class EmojiPickerMenu extends Component {
                     this.props.onEmojiSelected(this.state.filteredEmojis[this.state.highlightedIndex].code);
                 }
             };
-            document.addEventListener('keydown', this.keyDownHandler);
+
+            // Keyboard events are not bubbling in RN-Web
```
Comment (roryabraham) [inline src/pages/home/report/EmojiPickerMenu/index.js:109]: This comment isn't super helpful in it's current state – if you include it I think it's better to explain _why_ bubbling in this case is necessary for this feature to work.

**Review by roryabraham — CHANGES_REQUESTED** [review]
Code looks good, but a few other feature notes:

1. When search is refocused, we should highlight the string so that the old search will be replaced by anything new you type.
1. When you start typing again (any key other than arrow or enter), then the search should refocus.

This one is more of a nice-to-have, but if you:

1. Hover over an emoji to highlight it.
1. Use the down-arrow to focus the emoji,
1. Then the focus will go to the one that you have hovered, which feels a bit weird to me. I think it should just highlight the first emoji in the first row.

**Comment by parasharrajat** [comment]
> Then the focus will go to the one that you have hovered, which feels a bit weird to me. I think it should just highlight the first emoji in the first row.

@jasperhuangg Any input for this as you originally created the Picker.

**Comment by roryabraham** [comment]
I believe @stitesExpensify originally created the emoji picker. That last item I feel less strongly about too, so if you disagree let's just start with the first two.

**Comment by roryabraham** [comment]
My bad ... didn't mean to close this at all.

**Comment by parasharrajat** [comment]
@roryabraham 1, 2 is implemented and PR is updated. Thanks.

**Comment by jasperhuangg** [comment]
> > Then the focus will go to the one that you have hovered, which feels a bit weird to me. I think it should just highlight the first emoji in the first row.
> 
> @jasperhuangg Any input for this as you originally created the Picker.

So I actually implemented something that disables mouseover input for the `EmojiPickerMenu` when an arrow key is pressed, and reenables it again when the user moves their mouse. This should prevent the issue that @roryabraham is describing here.

File: src/pages/home/report/EmojiPickerMenu/index.js [hunk 2]
```diff
@@ -98,15 +98,28 @@ class EmojiPickerMenu extends Component {
...
+                    this.searchInput.focus();
+
+                    // Re-enable selection on the searchInput
+                    this.setState({selectTextOnFocus: true});
                 }
             };
 
-            // Keyboard events are not bubbling in RN-Web
+            // Keyboard events are not bubbling on TextInput in RN-Web, Bubbling was needed for this event to trigger
+            // event is attached to document. To fix this attach event in Capture phase.
```
Comment (roryabraham) [inline src/pages/home/report/EmojiPickerMenu/index.js:122]:
```suggestion
            // event handler attached to document root. To fix this, trigger event handler in Capture phase.
```

**Review by roryabraham — CHANGES_REQUESTED** [review]
Looks good and tests well. Just want to clean up the comment one more time before we merge.

**Comment by parasharrajat** [comment]
Updated.

**Comment by parasharrajat** [comment]
@roryabraham Gentle bump. Thanks

**Review by roryabraham — APPROVED** [review]

**Pull request merged**